pkg/certmaker/template.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package certmaker provides template parsing and certificate generation functionality
+// for creating X.509 certificates from JSON templates per RFC3161 standards.
+package certmaker
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"os"
+	"time"
+)
+
+// CertificateTemplate defines the structure for the JSON certificate templates
+type CertificateTemplate struct {
```

Comment:
Since the PR description mentioned smallstep, what was the motivation for creating your own template vs using smallstep's https://smallstep.com/docs/step-ca/templates/index.html?

Reply:
i'm actually using their template syntax w/ "go.step.sm/crypto/x509util" to parse those templates.

Reply:
i'm now using x509util's `TemplateData`.